src/OutputManager.C

```diff
@@ -0,0 +1,229 @@
+#endif
+    } else {
+      indexVec_[i] = ioBroker->create_output_mesh(
+        oname, stk::io::WRITE_RESULTS, *outputInfo->outputPropertyManager_);
+      // create selector and limit output if necessary
+      if (!outputInfo->targetNames_.empty()) {
+        // extract part
+        auto& partNameList = outputInfo->targetNames_;
+        stk::mesh::PartVector searchParts;
+        for (size_t k = 0; k < partNameList.size(); ++k) {
+          stk::mesh::Part* thePart = metaData->get_part(partNameList[k]);
+          if (NULL != thePart)
+            searchParts.push_back(thePart);
+          else
+            throw std::runtime_error(
+              "OutputManager::create_output_mesh: Part is null" +
+              partNameList[k]);
+        }
+
+        // selector and bucket loop
+        stk::mesh::Selector s_locally_owned =
+          metaData->locally_owned_part() & stk::mesh::selectUnion(searchParts);
+        // restrict output to the selector and entity rank
+        ioBroker->set_output_selector(
+          indexVec_[i], get_entity_rank(outputInfo, metaData), s_locally_owned);
+      }
+    }
+
+    // Tell stk_io how to output element block nodal fields:
+    // if 'true' passed to function, then output them as nodeset fields;
+    // if 'false', then output as nodal fields (on all nodes of the mesh,
+    // zero-filled) The option is provided since some
+    // post-processing/visualization codes do not correctly handle nodeset
+    // fields.
+    ioBroker->use_nodeset_for_part_nodes_fields(
+      indexVec_[i], outputInfo->outputNodeSet_);
+
+    // FIXME: add_field can take user-defined output name, not just varName
+    for (std::set<std::string>::iterator itorSet =
+           outputInfo->outputFieldNameSet_.begin();
+         itorSet != outputInfo->outputFieldNameSet_.end(); ++itorSet) {
+      std::string varName = *itorSet;
+      stk::mesh::FieldBase* theField =
+        stk::mesh::get_field_by_name(varName, *metaData);
+      if (NULL == theField) {
+        NaluEnv::self().naluOutputP0()
+          << " Sorry, no field by the name " << varName << std::endl;
+      } else {
+        // 'varName' is the name that will be written to the database
+        // For now, just using the name of the stk field
+        ioBroker->add_field(indexVec_[i], *theField, varName);
+      }
+    }
+  }
+}
+
+void
+OutputManager::perform_outputs(
+  const int timeStepCount,
+  const double currentTime,
+  stk::io::StkMeshIoBroker* ioBroker,
+  stk::mesh::MetaData* metaData,
+  const double wallTimeStart)
+{
+  const int nOutputs = infoVec_.size();
+  for (int i = 0; i < nOutputs; i++) {
+    auto* outputInfo = &(infoVec_[i]);
+    const auto resultsFileIndex = indexVec_[i];
+    const int modStep = timeStepCount - outputInfo->outputStart_;
+    const double elapsedWallTime = stk::wall_time() - wallTimeStart;
+    // find the max over all core
+    double g_elapsedWallTime = 0.0;
+    stk::all_reduce_max(
```

Comment:
Probably should do this check just once for all output blocks

Reply:
ah yeah thanks.  I'll pull it out of the loop